flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/ValueLiteralExpression.java

```diff
@@ -85,90 +97,105 @@ public boolean isNull() {
 			return Optional.empty();
 		}
 
-		final Class<?> valueClass = value.getClass();
-
 		Object convertedValue = null;
 
 		if (clazz.isInstance(value)) {
 			convertedValue = clazz.cast(value);
+		} else if (clazz == Period.class) {
+			convertedValue = convertToPeriod(value);
+		} else if (clazz == Duration.class) {
+			convertedValue = convertToDuration(value);
+		} else if (clazz == LocalDate.class) {
+			convertedValue = convertToLocalDate(value);
+		} else if (clazz == LocalTime.class) {
+			convertedValue = convertToLocalTime(value);
+		} else if (clazz == LocalDateTime.class) {
+			convertedValue = convertToLocalDateTime(value);
+		} else if (clazz == OffsetDateTime.class) {
+			convertedValue = convertToOffsetDateTime(value);
+		} else if (clazz == Instant.class) {
+			convertedValue = convertToInstant(value);
+		} else if (clazz == BigDecimal.class) {
+			convertedValue = convertToBigDecimal(value);
 		}
 
-		else if (valueClass == Duration.class && clazz == Long.class) {
-			final Duration duration = (Duration) value;
-			convertedValue = duration.toMillis();
-		}
+		return Optional.ofNullable((T) convertedValue);
+	}
 
-		else if (valueClass == Long.class && clazz == Duration.class) {
-			final Long longVal = (Long) value;
-			convertedValue = Duration.ofMillis(longVal);
+	private @Nullable LocalDate convertToLocalDate(Object value) {
+		Class<?> valueClass = value.getClass();
+		if (valueClass == java.sql.Date.class) {
+			return ((Date) value).toLocalDate();
+		} else if (valueClass == Integer.class) {
```

Comment:
include primitives everywhere?

Reply:
It cannot be a primitive, can it?

Reply:
yes, you are right. we are not using conversion class but the class of `Object`